autoretrieve.go

```diff
@@ -134,45 +133,51 @@ func New(cctx *cli.Context, dataDir string, cfg Config) (*Autoretrieve, error) {
 		return nil, fmt.Errorf("wallet initialization failed: %w", err)
 	}
 
-	walletAddr, err := wallet.GetDefault()
-	if err != nil {
-		logger.Warnf("Could not load any default wallet address, only free retrievals will be attempted: %v", err)
-		walletAddr = address.Undef
-	} else {
-		logger.Infof("Using default wallet address %s", walletAddr)
+	// Set up the PayChannelManager
+	ctx, shutdown := context.WithCancel(context.Background())
+	mpusher := filclient.NewMsgPusher(api, wallet)
```

Comment:
so we're not actually getting rid of filclient here, we're still using it to handle messages needed by the payment channel manager; is that a better state than outsourcing payment channels to filclient still? seems like either way we're in a half-way position.

the message pusher is just interacting with the api, so maybe we just copy that across too? https://github.com/application-research/filclient/blob/master/msgpusher.go#L25

basically, if we're trying to remove filclient, shouldn't we want to see it disappear from imports and go.mod?

Reply:
Latest commit copies the filclient MsgPusher and Keystore into Autoretrieve.